demo/demo.js

```diff
@@ -56,20 +78,24 @@ document.querySelector('#print').addEventListener('click', async () => {
   console.log('spec', spec);
   specEl.innerHTML = JSON.stringify(spec, null, '  ');
 
-  const report = await requestReport(MFP_URL, spec);
+  report = await requestReport(MFP_URL, spec);
   console.log('report', report);
   reportEl.innerHTML = JSON.stringify(report, null, '  ');
 
-  await getDownloadUrl(MFP_URL, report, 1000).then(
-    (url) => {
-      resultEl.innerHTML = url;
-      document.location = url;
-      return url;
-    },
-    (error) => {
-      console.log('result', 'error', error);
-      resultEl.innerHTML = error;
-      return error;
-    },
-  );
+  await getDownloadUrl(MFP_URL, report, 1000)
+    .then(
+      (url) => {
+        resultEl.innerHTML = url;
+        document.location = url;
+        return url;
+      },
+      (error) => {
+        console.log('result', 'error', error);
+        resultEl.innerHTML = error;
+        return error;
+      },
+    )
+    .then(() => {
+      report = null;
```

Comment:
In case of error, it will not pass here I think. Maybe it's better with a `finally`. (to test, it's easy with bad parameters, like removing the mandatory `datasource` at new line 69)